### DISCUSSION TITLE
rest api 403 error

### DISCUSSION THREAD

**Discussion by zuseola** [ant-media/Ant-Media-Server#5799] [FAQs]
https://antmedia.io/how-to-make-secure-ant-media-api-requests-to-web-panel/

hello :)
As described on this page, I have accurately created the JWT token, but I continue to encounter a 403 error. How can I recheck it?

I'll be waiting for your response , thanks!

**Comment by Mohit-3196** [comment]
Hi @zuseola,
How are you doing?

403 Forbidden error typically comes when there is an authentication issue or the token is not right.
Can you please recheck the steps? If that still doesn't work, I can probably make some guide for you or we can check this over a call.

Thank you.

**Reply by zuseola** [reply]
hello~ @Mohit-3196 
Thank you. I am using all services well except for the issue. Thank you for your response. Is there any mistake in my process?

![aaa1](https://github.com/ant-media/Ant-Media-Server/assets/129741799/cd7fca69-968e-4e6d-822d-246e877d0c03)

and restart antmedia..

![aaa2](https://github.com/ant-media/Ant-Media-Server/assets/129741799/c6d2cd3f-75fc-44ed-b9d2-e6c1a2270e5e)


![aaa3](https://github.com/ant-media/Ant-Media-Server/assets/129741799/414b8e54-b223-4589-b965-864c4e70e6c7)


thanks!

**Reply by Mohit-3196** [reply]
Hi @zuseola,
Did you follow the process under step 1?
edit the properties file located at `/usr/local/antmedia/conf/red5.properties` and edit the following settings:

```
server.jwtServerControlEnabled=true
server.jwtServerSecretKey=[your-secret-key-at-least-32-character]
```
You’ll need to make sure to set the` jwtServerControlEnabled` to true and enter a random 32-character key for jwtServerSecretKey.

Once you’ve done that, you’ll need to restart the Ant Media Server service:

`sudo service antmedia restart`

**Reply by zuseola** [reply]
thanks for your answer , but i already done 
If we exclude that part, is it still considered normal to succeed?
thanks!

**Reply by Mohit-3196** [reply]
Hi @zuseola.

> but i already done

I see. Can you tell me which Ant Media Server version are you using and let me try it on my end?

> If we exclude that part, is it still considered normal to succeed?
No, the first step is important for it to work.

**Reply by zuseola** [reply]
thanks for your answer @Mohit-3196 

Ant Media server version

{
    "versionName": "2.5.3",
    "versionType": "Enterprise Edition",
    "buildNumber": "20230107_0851"
}

**Reply by Mohit-3196** [reply]
Hi @zuseola ,
Can you please update your Ant Media Server to the latest version 2.7.0? That'll resolve the issue when using JWT for both application and management API calls.
Here's how you can upgrade the version  https://antmedia.io/docs/guides/installing-on-linux/upgrading-ant-media-server/

Let me know after you have upgraded. If you need help with the upgrade, I'll be here.
Thank you

**Reply by zuseola** [reply]
{
    "versionName": "2.7.0",
    "versionType": "Enterprise Edition",
    "buildNumber": "20231031_0626"
}

I have upgraded the version, but I am still experiencing a 403 error, making the REST API inaccessible. It seems like I might be doing something wrong :(

**Reply by zuseola** [reply]
Is it possible to use REST API without having to use JWT tokens or my account? Can I make the REST API accessible without the need for authentication using JWT tokens or an account?

**Comment by mitchellalderson** [comment]
Hi there, I just finished implementing an API proxy for my Ant server that uses JWT, one thing that I noticed is that Ant does not want a prefix with the token when making the REST call. ie: `bearer <jwt>` vs. just `<jwt>`. I would check your send log, as I think Postman might add `bearer` to the token automatically.

Also, I was just using a plain `Authorization` header, not sure if you have something custom that would require the header key to be `ProxyAuthorization`?

**Reply by zuseola** [reply]
@farawaysouthwest 
thank for your answer

How do I send a CURL API request then? Is this format correct?

curl -k -X GET -H "Content-Type: application/json" -H "ProxyAuthorization:eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.RyqIBdYTJcE8umPkoSlwpji1ZM5rDDcg7lzwIe-bk_w" "https://mydomain:5443/rest/v2/system-resources"

**Reply by mitchellalderson** [reply]
Hi @zuseola,

I'm able to successfully query my community 2.6.4 server with the following command:
```
curl --request GET \
  --url https://<server URL>:5443/LiveApp/rest/v2/<api route> \
  --header 'Authorization: <jwt key>'
```
Take note in my header name and format, that could be your issue? Also, make sure that you are forming the JWT with the signing secret correctly, that can take some trial and error.

**Reply by zuseola** [reply]
thanks for your answer @farawaysouthwest 
I am still encountering a 403 error. Is there something wrong with my process?

![image_2023-11-17_12-02-30](https://github.com/ant-media/Ant-Media-Server/assets/129741799/d6a2a022-de22-408c-a93f-fcd1e7c8623a)

vi red5-web.properties
jwtControlEnabled=true
jwtSecretKey=xUe63IlI2vhK1nf8RtZBxZzQ5TTwZqUn

systemctl restart antmedia.service

![image_2023-11-17_12-01-57](https://github.com/ant-media/Ant-Media-Server/assets/129741799/08cda747-d285-46bc-97fe-41e2b05b2964)


curl --request GET \
  --url https://mydomain:5443/LiveApp/rest/v2/applications?null \
  --header 'Authorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.RyqIBdYTJcE8umPkoSlwpji1ZM5rDDcg7lzwIe-bk_w'

**Reply by mitchellalderson** [reply]
Hi there! I think if this is a remote machine and you have JWT enabled, you should disable the IP filter entirely; with your current setup, only `localhost` is allowed to query REST.

**Comment by Mohit-3196** [comment]
Hi @zuseola,
Let me help you with some screenshots.

1.  Edit the `red5.properties` located under `/usr/local/antmedia/conf`
2. Make the changes as shown in the screenshot. You can use any random 32 characters as the jwtServerSecretKey
![Screenshot 2023-11-17 101029](https://github.com/ant-media/Ant-Media-Server/assets/86982446/57407de5-7b9b-4d06-afac-4a0e7eeec69b)

3. After making the changes, save them and restart your Ant Media Server
`sudo service antmedia restart`

4. Enter the secret key and generate your token
![image](https://github.com/ant-media/Ant-Media-Server/assets/86982446/b0bd90d3-ded4-4bed-833e-f4b623abefcc)

5. Now, you can make the REST API call
`curl -X GET -H "Content-Type: application/json" -H "ProxyAuthorization:eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.qxAyw9lZTg_qdI0abEzpEQ-DRkVM9qqCbZ6ZfJ5gpUA" "http://65.0.135.239:5080/rest/v2/system-resources"`
![image](https://github.com/ant-media/Ant-Media-Server/assets/86982446/25ddf664-e7ce-4fd6-ac70-40fe77c1d244)


> Also, you do not have to enable `JWT Filter for RESTful APIs` in the application settings as that's for the [application level API calls](https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/jwt-rest-api-filter/) and not management API calls

In case it still doesn't work for you, let me know and we can schedule a call and check this together.

Thank you
